src/Validation/DonationValidator.php

```diff
@@ -54,13 +57,61 @@ public function validate( Donation $donation ): ValidationResult {
 	}
 
 	public function needsModeration( Donation $donation ): bool {
-		// TODO: add TextPolicyValidator
 		$violations = [];
 
 		$violations[] = $this->amountPolicyValidator->validate( $donation->getAmount(), $donation->getInterval() );
 
+		$violations = array_merge(
+			$violations,
+			$this->getBadWordViolations( $donation )
+		);
+
 		$this->policyViolations = array_filter( $violations );
+
 		return !empty( $this->policyViolations );
 	}
 
+	public function getPolicyViolations() {
+		return $this->policyViolations;
+	}
+
+	private function getBadWordViolations( Donation $donation ) {
+		$violations = [];
+
+		$flags = TextPolicyValidator::CHECK_BADWORDS |
+			TextPolicyValidator::IGNORE_WHITEWORDS |
+			TextPolicyValidator::CHECK_URLS;
+		$fieldTextValidator = new FieldTextPolicyValidator( $this->textPolicyValidator, $flags );
+		$personalInfo = $donation->getPersonalInfo();
+
+		if ( $personalInfo ) {
+			$violations[] = $this->getFieldViolation(
+				$fieldTextValidator->validate( $personalInfo->getPersonName()->getFirstName() ),
+				'vorname'
+			);
+			$violations[] = $this->getFieldViolation(
+				$fieldTextValidator->validate( $personalInfo->getPersonName()->getLastName() ),
+				'nachname'
+			);
+			$violations[] = $this->getFieldViolation(
+				$fieldTextValidator->validate( $personalInfo->getPersonName()->getCompanyName() ),
+				'firma'
+			);
+			$violations[] = $this->getFieldViolation(
+				$fieldTextValidator->validate( $personalInfo->getPhysicalAddress()->getStreetAddress() ),
+				'strasse'
+			);
+			$violations[] = $this->getFieldViolation(
+				$fieldTextValidator->validate( $personalInfo->getPhysicalAddress()->getPostalCode() ),
+				'plz'
+			);
+			$violations[] = $this->getFieldViolation(
+				$fieldTextValidator->validate( $personalInfo->getPhysicalAddress()->getCity() ),
+				'ort'
+			);
+		}
+
+		return $violations;
```

Comment:
If you do `return [ ... ]`, you can avoid having the local var and all the assignments to it.